File: tests/basic_enum.lua
local unit = Test.Unit
local tuple = Test.Tuple("test", 20)
local named = Test.Named({ name = "test", int = 20 })

assert(unit.unit)
assert(tuple.tuple[1] == "test")
assert(tuple.tuple[2] == 20)
assert(named.named.name == "test")
assert(named.named.int == 20)
local new_tuple = tuple.tuple
new_tuple[1] = "hey"
tuple.tuple = new_tuple
assert(tuple.tuple[1] == "hey")
assert(tuple.tuple[2] == 20)

File: tests/basic_enum.rs
use mlua_gen::mlua_gen;

#[test]
pub fn test() {
    #[mlua_gen]
    pub(crate) enum Ab {
        Unit,
        Tuple(String, i32),
        Named { name: String, int: i32 },
    }

    let lua = mlua::Lua::new();
    lua.globals()
        .set("Test", Ab::Tuple(String::from("uwu"), 42))
        .unwrap();

    lua.load(include_str!("./basic_enum.lua")).exec().unwrap();
}
